lib/tests/test_dynamic_window.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2019 One Identity
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "dynamic-window.h"
+#include <criterion/criterion.h>
+
+Test(dynamic_window, window_stat_reset)
+{
+  DynamicWindow win;
+  dynamic_window_stat_reset(&win.stat);
+  cr_expect_eq(win.stat.sum, 0);
+  cr_expect_eq(win.stat.n, 0);
+}
+
+Test(dynamic_window, window_stat_reset_when_counter_is_set)
```

Comment:
`counter` should be renamed to `pool`. (4/7)

Reply:
This remained the same after the fixup commits. Also 3 lines below, instead of `dynamic_window_set_counter()`, the function `dynamic_window_set_pool()` should be used,